### ISSUE TITLE
Can I get some evaluating metric or something else?

### ISSUE THREAD

**Issue by Asuna88** [vinthony/deep-blind-watermark-removal#4]
Excuse me , can I know how to evaluate the result of removing watermarks?

Do you have some metric like mAP or IoU?  I just wonder how to evaluate the results of removing watermarks.

Can I get some evaluating metric or something else?

Thx!

**Comment by vinthony** [comment]
Hi, I think mAP and IoU more likely works for watermark detection other than removal.

Currently, we evaluate the methods by the standard image restoration metrics, such as PSNR, SSIM, and LPIPS.

**Comment by Asuna88** [comment]
> Hi, I think mAP and IoU more likely works for watermark detection other than removal.
> 
> Currently, we evaluate the methods by the standard image restoration metrics, such as PSNR, SSIM, and LPIPS as metioned in the paper.


 Got it. 
So if the metrics like PSNR is large enough,then the effect is good? 
or SSIM which in [0,1] is large enough,then the effect is good? 




What about the datasets that don't have  GT images(for example, I pick some other pictures at random which are not part of my original dataseta)? 

You can not cacluate the PSNR or SSIM, right? In that case, how do you evaluate the effect of watermark removing?

Thanks a lot!

**Comment by Asuna88** [comment]
> Hi, I think mAP and IoU more likely works for watermark detection other than removal.
> 
> Currently, we evaluate the methods by the standard image restoration metrics, such as PSNR, SSIM, and LPIPS as metioned in the paper.


 Got it. 
So if the metrics like PSNR is large enough,then the effect is good? 
or SSIM which in [0,1] is large enough,then the effect is good? 




What about the datasets that don't have  GT images(for example, I pick some other pictures at random which are not part of my original dataseta)? 

You can not cacluate the PSNR or SSIM, right? In that case, how do you evaluate the effect of watermark removing?

Thanks a lot!

**Comment by vinthony** [comment]
Yes, larger PSNR and SSIM means better performance.

if you do not have the ground truth image, the PSNR and SSIM can not be evaluated.
 
In this case, I think an appropriate way is to evaluate the model via some pre-trained image quality assessment models, or watermark detection models.

**Comment by Asuna88** [comment]
> Yes, larger PSNR and SSIM means better performance.
> 
> if you do not have the ground truth image, the PSNR and SSIM can not be evaluated.
> 
> In this case, I think an appropriate way is to evaluate the model via some pre-trained image quality assessment models, or watermark detection models.

Thx so much!

The fact is I can get the PSNR and SSIM metrics in my test datasets since they have GT images.

The problem is I canot get PSNR and SSIM metrics in my other images which don't have 
GT images.  I just evaluated the effect of watermark removal with my naked eyes after the model processing, you know,  it's so inefficient for me. 

You mentioned "some pre-trained image quality assessment models", can you explain this model in detail (for example, what model it is, what algorithm and principle it is) ? Do your have  link of the paper ?

Thanks.

**Comment by vinthony** [comment]
Hi, I think we do not have a model for the quality of the watermark removal problem specifically.

However, there might some other ideas we can borrow.

For example, for the image composite problem, [RealismCNN](https://github.com/junyanz/RealismCNN) can be used to evaluate the performance of the composited images. Or we can evaluate the performance via the Blind Image Quality Assessment(IQA) models? here is a [survey](http://orca.cf.ac.uk/125265/1/IEEE%20Access%2008822415.pdf).

hope it helps.

**Comment by Asuna88** [comment]
> Hi, I think we do not have a model for the quality of the watermark removal problem specifically.
> 
> However, there might some other ideas we can borrow.
> 
> For example, for the image composite problem, [RealismCNN](https://github.com/junyanz/RealismCNN) can be used to evaluate the performance of the composited images. Or we can evaluate the performance via the Blind Image Quality Assessment(IQA) models? here is a [survey](http://orca.cf.ac.uk/125265/1/IEEE%20Access%2008822415.pdf).
> 
> hope it helps.


Sounds like a good idea.

Thank for your helpful suggestions and I am going to learn abot  the method " RealismCNN " and try it out.

thanks a lot!
